### ISSUE TITLE
Bug when commenting visually selected lines containing an empty commented line 

### ISSUE THREAD

**Issue by Hotschke** [tpope/vim-commentary#58]
I have encountered following bug when using commentary.vim to comment a block of python code:

```
x = 12
#
y = 12
```
Do a Visual line selection for the whole file (`set ft=python`) with `ggVG` and press `gc`. Nothing happens. Removing the `#` symbol without deleting the line `gc` works again.

This occurs as long as the commented line is empty or does only contain whitespace:
```
x = 12
# non empty line: everything works as expected
y = 12
```

I have tested this with gvim 7.4.873 on linux and neovim v0.1.1-15 under OSX.

**Comment by jonahx** [comment]
I've seen the same "nothing happens" bug when doing `gc` on the following visually selected block in my .vimrc:

```
let mapleader = ","
vnoremap <C-c> "*y
inoremap <C-v> <C-r><C-p>*
runtime macros/matchit.vim

"""""""""""""""""""""""""""""""""""
"  Vundle Plugins
"""""""""""""""""""""""""""""""""""
set nocompatible
set ruler
filetype off
```

**Comment by cdepillabout** [comment]
I'm running into this as well.  Similar symptoms as @jonahx and @Hotschke.

**Comment by githorse** [comment]
Agreed with above. To clarify, given the following lines:

    1 "
    2 foo
    3 " 
    4
    5 bar
    6

I have the following problems:

1. `gcap` fails when there is an empty commented line immediately above or below the paragraph. (`2Ggcap`).
2. Commenting with visual selection fails when the selection includes an empty commented line at the top (`1GVjgc`) or in the middle of the selection `1GV5jgc`. It works when the empty commented line is at the bottom (`2GVjgc`).

**Comment by chrisbra** [comment]
I think this was fixed with 73e0d9a9d1f51b6cc9

**Comment by Hotschke** [comment]
I can confirm that for my case it is fixed. The code changes to:
```
# x = 12
# #
# y = 12
```

Well, some might argue that the already commented line does not need a second comment character. However, this is a matter of a different discussion. I am closing the issue.
